### PR TITLE
MSA StatementFinder inputs and outputs

### PR DESCRIPTION
**Pull request by pagreene**
This PR adds methods to the MSA StatementFinder class to get input and output agents.

### CODE REVIEW

**Comment by bgyori** [comment]
I wasn't able to get this to work - there are some trivial issues like using keys() instead of items(), and referring to "entites" instead of "entities" but even after fixing those I am running into errors. Are you sure you didn't have additional unpushed commits?

File: bioagents/msa/msa.py [hunk 1]
```diff
@@ -162,6 +162,89 @@ def get_statements(self, block=None, timeout=10):
 
         return self._statements[:]
 
+    def get_fixed_agents(self):
+        """Get a dict of the agents that were used as inputs, keyed by role."""
+        raw_dict = {'subject': [self.query.subj], 'object': [self.query.obj],
+                    'other': self.query.agents}
+        ret_dict = {}
+        for role, ag_list in raw_dict.keys():
```
Comment (bgyori) [inline bioagents/msa/msa.py]: This should probably be items() here

File: bioagents/msa/msa.py [hunk 2]
```diff
@@ -162,6 +162,89 @@ def get_statements(self, block=None, timeout=10):
...
+            is used.
+        """
+        # Check to make sure role is valid.
+        if other_role not in ['subject', 'object', None]:
+            raise ValueError('Invalid role of type %s: %s'
+                             % (type(other_role), other_role))
+
+        # Get the namespace and id of the original entity.
+        if entities is None:
+            entities = set(self.query.entites.keys())
```
Comment (bgyori) [inline bioagents/msa/msa.py]: entites -> entities

**Comment by bgyori** [comment]
After much head scratching, I think I fixed some of the issues. I added a single test condition for a to_target query, but we should have similar tests for all other query types because it is possible that not all of those work yet.

**Comment by pagreene** [comment]
Sorry, I meant to write and run some tests, but I got distracted by Medscan stuff. It still doesn't seem to be working as far as I can tell. Working on it now.

**Comment by pagreene** [comment]
Alright, I figured it out. It is now working well on my system. I added some more tests for good measure.

**Comment by pagreene** [comment]
As a general note, you should make sure `block` is set to True when you use this. Otherwise if the request is still running, you will get nothing.

**Comment by bgyori** [comment]
This works well in the integration now! As an aside, with the sorting of other names, the evidence counts propagated into the HTML, and the removal of None subjects, the prioritization of the various lists (summary, provenance tab, HTML) line up almost perfectly which is a nice improvement.

**Comment by bgyori** [comment]
Tests are passing 👍

**Pull request merged**